### ISSUE TITLE
CreateSourceGroup.cmake expects path to be acceptable regex

### ISSUE THREAD

**Issue by FatalCatharsis** [biojppm/rapidyaml#206]
If you use cmake to attempt to initialize the project, it fails configuration in the event that the path to the project contains control characters in regex.

result of configuring project:

```
CMake Error at ext/c4core/cmake/CreateSourceGroup.cmake:19 (string):
  string sub-command REGEX, mode REPLACE failed to compile regex
  "F:/Documents/Projects/personal/c++/native-library-maker/libs/rapidyml/rapidyaml/ext/c4core/src".
Call Stack (most recent call first):
  ext/c4core/cmake/c4Project.cmake:1801 (create_source_group)
  ext/c4core/cmake/c4Project.cmake:1723 (c4_add_target)
  ext/c4core/CMakeLists.txt:83 (c4_add_library)
```

The folder "c++" is the culprit. If I rename that to "cpp", the project configures fine because that line in CreateSourceGroup.cmake compiles the path as a valid regex. Maybe update the cmake file to escape regex characters that are acceptable path characters like the "+" and "." and whatever else might be slipping my mind.

**Comment by biojppm** [comment]
Thanks for reporting.

**Comment by biojppm** [comment]
This was fixed in biojppm/cmake#11, merged to rapidyaml since ~2months ago.
